rotkehlchen/db/upgrades/v34_v35.py

```diff
@@ -50,13 +50,45 @@ def _rename_assets_identifiers(cursor: 'DBCursor') -> None:
     cursor.executemany('UPDATE OR IGNORE assets SET identifier=? WHERE identifier=?', sqlite_tuples)  # noqa: E501
 
 
+def _change_xpub_mappings_primary_key(write_cursor: 'DBCursor', read_cursor: 'DBCursor') -> None:
```

Comment:
To not have multiple cursors open I suggest passing only the read cursor in this function and instead opening and closing (using a with context) a read cursor to read the data from `xpub_mappings` at the start of this function

Reply:
Uh... I'm not sure that I got you, could you please give a bit of code?

Reply:
I had a typo but still should be easy to understand:

To not have multiple cursors open I suggest passing only the **write cursor** in this function and instead opening and closing (using a with context) a read cursor to read the data from xpub_mappings at the start of this function.

```python
with conn.read_ctx() as read_cursor:
    xpub_mappings = read_cursor.execute('SELECT * from xpub_mappings').fetchall()
...
...
...
```